### PR TITLE
[KernelGen][Nvidia] Add sym_stride operator with Triton kernel

### PR DESCRIPTION
**Pull request by bwbwzzz**
## Summary

Adds the `sym_stride` operator, which returns the stride of the input tensor as a tuple of integers. This operator is used for `torch.compile` symbolic tracing and returns tensor metadata rather than a computed tensor.

## Performance

Test command: `pytest benchmark/test_sym_stride.py --level core` (NVIDIA H20)

### float16
| Size Detail | Torch Latency (ms) | Gems Latency (ms) | Speedup |
|---|---|---|---|
| 2, 3 | 0.003 | 0.003 | 1.040 |
| 10, 20, 30 | 0.003 | 0.003 | 1.039 |
| 5, 10 | 0.003 | 0.003 | 1.000 |
| 100 | 0.003 | 0.003 | 1.010 |
| 1, 2, 3, 4 | 0.003 | 0.003 | 1.010 |

### float32
| Size Detail | Torch Latency (ms) | Gems Latency (ms) | Speedup |
|---|---|---|---|
| 2, 3 | 0.003 | 0.003 | 1.010 |
| 10, 20, 30 | 0.003 | 0.003 | 1.010 |
| 5, 10 | 0.003 | 0.003 | 1.000 |
| 100 | 0.003 | 0.003 | 1.000 |
| 1, 2, 3, 4 | 0.003 | 0.003 | 1.000 |

### bfloat16
| Size Detail | Torch Latency (ms) | Gems Latency (ms) | Speedup |
|---|---|---|---|
| 2, 3 | 0.003 | 0.003 | 1.000 |
| 10, 20, 30 | 0.003 | 0.003 | 1.000 |
| 5, 10 | 0.003 | 0.003 | 1.000 |
| 100 | 0.003 | 0.003 | 1.000 |
| 1, 2, 3, 4 | 0.003 | 0.003 | 1.000 |

**Arithmetic Mean Speedup: 1.011x**

## Testing

- Parametrized tests over 5 shapes and float16/float32/bfloat16 dtypes (15 cases)
- Validated against PyTorch reference `torch.ops.aten.sym_stride`
- Tested on: NVIDIA H20, CUDA 13.2

## Multi-backend Testing

| Backend | Accuracy Test | Benchmark | Speedup (mean) | Notes |
|---|---|---|---|---|
| Nvidia (H20) | PASS | PASS (15 cases, --level core) | 1.011x | Primary |
| Tianshu | — | — | — | — |
| Muxi | — | — | — | — |
| Ascend | — | — | — | — |
| Hygon | — | — | — | — |

## Files Changed

- `src/flag_gems/ops/sym_stride.py`: Operator implementation (returns tensor stride metadata)
- `tests/test_sym_stride.py`: Accuracy test
- `benchmark/test_sym_stride.py`: Performance benchmark
- `src/flag_gems/ops/__init__.py`: Register import and `__all__`
- `src/flag_gems/__init__.py`: Register to `_FULL_CONFIG`
- `conf/operators.yaml`: Add operator entry (kind: Utility, stage: alpha 5.4)

### CODE REVIEW

File: benchmark/test_sym_stride.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
+import pytest
+import torch
+
+from . import base, consts
+
+# Benchmark shapes for sym_stride - covering various tensor dimensionalities
+SYM_STRIDE_SHAPES = [(2, 3), (10, 20, 30), (5, 10), (100,), (1, 2, 3, 4)]
+
+
+def _sym_stride_input_fn(shape, cur_dtype, device):
```
Comment (tengqm) [inline benchmark/test_sym_stride.py]: What the hell is this function used for?

File: conf/operators.yaml [hunk 2]
```diff
@@ -7199,6 +7199,17 @@ ops:
...
       - stable: '5.0'
+  - id: sym_stride
+    description: Returns the stride of the input tensor as a tuple of integers. Used for torch.compile symbolic tracing.
+    for:
+      - sym_stride
+    labels:
+      - aten
+      - KernelGen
+    kind:
+      - Utility
```
Comment (tengqm) [inline conf/operators.yaml]:
```suggestion
      - Tensor
```

File: src/flag_gems/ops/sym_stride.py [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    return 1
+
+
+def sym_stride(A):
+    """Returns the stride of the input tensor.
+
+    This operator returns the stride of the input tensor as a list of integers.
+    It is used for torch.compile symbolic tracing.
+    """
+    logger.debug("GEMS sym_stride")
```
Comment (tengqm) [inline src/flag_gems/ops/sym_stride.py]: Why suddenly lower case the ID of the operator, after SO MANY rounds of reviews?

File: src/flag_gems/ops/sym_stride.py [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    # sym_stride returns tensor metadata (stride), not a computed tensor
+    # For FlagGems integration, we return the actual stride
+    return A.stride()
+
+
+def sym_stride_(A):
+    """In-place version of sym_stride (returns the same stride, tensor is unchanged)."""
+    logger.debug("GEMS sym_stride_")
```
Comment (tengqm) [inline src/flag_gems/ops/sym_stride.py]: Here as well...

File: src/flag_gems/ops/sym_stride.py [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+    return A.stride()
```
Comment (tengqm) [inline src/flag_gems/ops/sym_stride.py]: How come this is called an in-place operator if A is not changed at all?

File: tests/test_sym_stride.py [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+
+import flag_gems
+
+from . import accuracy_utils as utils
+
+# Test shapes for sym_stride - covering various tensor dimensionalities
+SYM_STRIDE_SHAPES = [(2, 3), (10, 20, 30), (5, 10), (100,), (1, 2, 3, 4)]
+
+
+@pytest.mark.sym_stride
```
Comment (tengqm) [inline tests/test_sym_stride.py:12]: We add a 'sym_stride_' operator and we don't test it?

File: src/flag_gems/__init__.py [hunk 7]
```diff
@@ -623,6 +623,7 @@ def torch_ge(v):
     ("sum.dim_IntList", sum_dim),
     ("sum.out", sum_out),
     ("svd", svd),
+    ("sym_stride", sym_stride),
```
Comment (tengqm) [inline src/flag_gems/__init__.py:626]: We add an in-place operator but do not register it?

File: conf/operators.yaml (hunk 2 above)
Comment (bwbwzzz) [reply conf/operators.yaml]: i have revised the mistake thks

File: src/flag_gems/ops/sym_stride.py (hunk 5 above)
Comment (bwbwzzz) [reply src/flag_gems/ops/sym_stride.py]: this function should not exist，i have removed， thks

File: src/flag_gems/__init__.py (hunk 7 above)
Comment (bwbwzzz) [reply src/flag_gems/__init__.py:626]: have removed

**Review by tengqm — APPROVED** [review]

**Review by bin913 — APPROVED** [review]
